### PR TITLE
[WIP] linter for order of imports for kubeadm

### PR DESCRIPTION
**Pull request by dims**
cc @neolit123 

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by xinydev** [comment]
Maybe we should add empty lines as separators between sections, otherwise `gofmt` will reformat these import lines.

before `gofmt`
```
"sort"
"strings"
"time"
v1 "k8s.io/api/core/v1"
metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
bootstrapapi "k8s.io/cluster-bootstrap/token/api"
bootstraputil "k8s.io/cluster-bootstrap/token/util"
bootstrapsecretutil "k8s.io/cluster-bootstrap/util/secrets"
"github.com/pkg/errors"
```

after `gofmt`
```
"github.com/pkg/errors"
"k8s.io/api/core/v1"
"k8s.io/apimachinery/pkg/apis/meta/v1"
metav1 "k8s.io/client-go/applyconfigurations/meta/v1"
bootstrapapi "k8s.io/cluster-bootstrap/token/api"
bootstraputil "k8s.io/cluster-bootstrap/token/util"
bootstrapsecretutil "k8s.io/cluster-bootstrap/util/secrets"
"sort"
"strings"
"time"
```

File: cmd/orderimports/orderimports.go [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+			var originalImports []string
+			for _, imp := range file.Imports {
+				importPath := strings.Replace(imp.Path.Value, "\"", "", -1)
+				parts := strings.Split(importPath, "/")
+				originalImports = append(originalImports, importPath)
+				// standard library imports are first
+				if !strings.Contains(parts[0], ".") {
+					stdlibImports = append(stdlibImports, importPath)
+				} else {
+					if strings.Contains(parts[0], "k8s.io/kubernetes") {
```
Comment (xinydev) [inline cmd/orderimports/orderimports.go:99]: maybe
```
parts[0] == "k8s.io" && parts[1] == "kubernetes"
```
or
```
strings.HasPrefix(importPath, "k8s.io/kubernetes")
```

Comment (xinydev) [reply cmd/orderimports/orderimports.go:99]: and we should skip the `zz_generated` files?

**Comment by neolit123** [comment]
@xinydev are you willing to continue the work on such a linter?

Note we are in code freeze for 1.22, thought.

**Comment by xinydev** [comment]
> @xinydev are you willing to continue the work on such a linter?
> 
> Note we are in code freeze for 1.22, thought.

I am very happy to continue the work on such a linter

**Pull request closed without merging**